dialects/autodiff/auxiliary/autodiff_aux.cpp

```diff
@@ -146,7 +146,7 @@ const Def* zero_def(const Def* T) {
 const Def* op_sum(const Def* T, DefArray defs) {
     // TODO: assert all are of type T
     auto& world = T->world();
-    return world.raw_app(world.raw_app(world.ax<sum>(), {world.lit_nat(defs.size()), T}), world.tuple(defs));
+    return world.app<false>(world.app<false>(world.ax<sum>(), {world.lit_nat(defs.size()), T}), world.tuple(defs));
```

Comment:
We want to normalize at this point.
The sum should be resolved to a chain of additions.

I think that most of the following `<false>` should be `<true>` in the general spirit of "always apply normalizers" (except the normalizer pass-through cases).

Reply:
So, you are telling me, the original code should have used `app` instead of `raw_app`?

Reply:
Probably. I think I misunderstood `raw_app` and some of my code is broken/does not work as intended.